src/main/scala/org/hammerlab/guacamole/assembly/DeBruijnGraph.scala

```diff
@@ -213,13 +215,20 @@ class DeBruijnGraph(val kmerSize: Int,
       // Check if the source node was merged into the current one
       lazy val foundMergedSink = nodeContainingSink.exists(_._1 == next)
       val foundSink = (next == sink || foundMergedSink)
-      if (!foundSink && currentPath.size < maxPathLength) {
+      val nextNodes = children(next)
+      val filteredNextNodes = (if (avoidLoops) nextNodes.filterNot(visited.contains) else nextNodes)
+      if (!foundSink && filteredNextNodes.nonEmpty && currentPath.size < maxPathLength) {
+
+        // Track if this is a branching node
+        if (filteredNextNodes.size > 1) {
+          lastBranchIndex = currentPath.length
+        }
+
         // Keep searching down tree
-        val nextNodes = children(next)
-        frontier ++= (if (avoidLoops) nextNodes.filterNot(visited.contains) else nextNodes)
+        frontier.pushAll(filteredNextNodes)
 
       } else {
-        //found sink or too long
+        // Found sink or too long
         if (foundSink && currentPath.size + 1 >= minPathLength) {
```

Comment:
Maybe the nesting is confusing here, but what should happen is, we should end up here if we either found our goal or we cut-off the search. This check that if we did find the sink/goal, the path was long enough.  Will clarify some comments, but let me know if that seems wrong